libenv/sysinfo.c

```diff
@@ -1085,14 +1085,24 @@ static void OSClasses(EvalContext *ctx)
 
     if (os_release_json != NULL)
     {
-        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "os_release", os_release_json, CF_DATA_TYPE_CONTAINER, "source=agent,os,release,derived_from=/etc/os-release");
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "os_release",
+                                      os_release_json, CF_DATA_TYPE_CONTAINER,
+                                      "source=agent,os,release,derived_from=/etc/os-release");
         const char *os_release_id = JsonObjectGetAsString(os_release_json, "ID");
-        if(os_release_id != NULL)
+        const char *os_release_version = JsonObjectGetAsString(os_release_json, "VERSION");
+        char os_release_flavor[256];
+        snprintf(os_release_flavor, 256, "%s_%s", os_release_id, os_release_version);
```

Comment:
If they are NULL, we cannot access the os-release file. We still do all the other detection code, but won't be able to determine and set coreos class.